File: src/use-cases/interfaces/send-direct-message.ts
import { IDirectMessageEntity } from "../../entities";
import { IUseCase } from "../types";

export interface ISendDirectMessageUseCase extends IUseCase {
  execute(
    senderCognitoId: string,
    recipientCognitoId: string,
    text: string
  ): Promise<IDirectMessageEntity>;
}

File: lib/src/use-cases/interfaces/send_direct_message.dart
import 'package:dartz/dartz.dart';

import '../../entities/entities.dart';

abstract class ISendDirectMessageUseCase {
  Future<Either<Tuple2<ConversationEntity, MessageEntity>, Failure>>
      sendDirectMessage(
    String recipientCognitoId,
    String message,
  );
}
